### PR TITLE
Feature/ninjemys

### PR DESCRIPTION
**Pull request by 0x72D0**
## Description
Merge the noetic branch.

### CODE REVIEW

**Comment by supertoto29** [comment]
@0x72D0 Looks like there is a problem with the L4T ARM64

File: .github/workflows/docker-image-develop.yml [hunk 1]
```diff
@@ -85,43 +85,43 @@ jobs:
...
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja run_tests
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja tests
+
+  #     - name: Create Docker Image Tag
+  #       run: |
+  #         docker tag build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} ${GITHUB_REMOTE_URL}/${IMAGE_NAME}:${ARCH}-${TARGET_TYPE}-${TARGET_VERSION}
+
+  #     - name: Push Image to Github Packages Registry
+  #       run: |
+  #         docker push --all-tags ${GITHUB_REMOTE_URL}/${IMAGE_NAME}
```
Comment (Kam0ss) [inline .github/workflows/docker-image-develop.yml:124]: This code should either be removed or not commented.

File: .github/workflows/docker-image-feature.yml [hunk 2]
```diff
@@ -85,43 +85,43 @@ jobs:
...
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja run_tests
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja tests
+
+  #     - name: Create Docker Image Tag
+  #       run: |
+  #         docker tag build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} ${GITHUB_REMOTE_URL}/${IMAGE_NAME}:${ARCH}-${TARGET_TYPE}-${TARGET_VERSION}-${GITHUB_REF##*/}
+
+  #     - name: Push Image to Github Packages Registry
+  #       run: |
+  #         docker push --all-tags ${GITHUB_REMOTE_URL}/${IMAGE_NAME}
```
Comment (Kam0ss) [inline .github/workflows/docker-image-feature.yml:124]: This code should either be removed or uncommented.

File: .github/workflows/docker-image-fix.yml [hunk 3]
```diff
@@ -85,43 +85,43 @@ jobs:
...
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja run_tests
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja tests
+
+  #     - name: Create Docker Image Tag
+  #       run: |
+  #         docker tag build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} ${GITHUB_REMOTE_URL}/${IMAGE_NAME}:${ARCH}-${TARGET_TYPE}-${TARGET_VERSION}-${GITHUB_REF##*/}
+
+  #     - name: Push Image to Github Packages Registry
+  #       run: |
+  #         docker push --all-tags ${GITHUB_REMOTE_URL}/${IMAGE_NAME}
```
Comment (Kam0ss) [inline .github/workflows/docker-image-fix.yml:124]: This code should either be removed or uncommented.

File: .github/workflows/docker-image-master.yml [hunk 4]
```diff
@@ -91,46 +91,46 @@ jobs:
...
+  #         docker run build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} catkin_make --use-ninja tests
+
+  #     - name: Create Docker Image Tag
+  #       run: |
+  #         docker tag build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} ${GITHUB_REMOTE_URL}/${IMAGE_NAME}:${ARCH}-${TARGET_TYPE}-${TARGET_VERSION}
+  #         docker tag build-${TARGET_VERSION}-${GITHUB_REF##*/}-${GITHUB_RUN_NUMBER} ${GITHUB_REMOTE_URL}/${IMAGE_NAME}:${ARCH}-${TARGET_TYPE}-latest
+
+  #     - name: Push Image to Github Packages Registry
+  #       run: |
+  #         docker push --all-tags ${GITHUB_REMOTE_URL}/${IMAGE_NAME}
```
Comment (Kam0ss) [inline .github/workflows/docker-image-master.yml:133]: This code should either be removed or uncommented.

**Review by Kam0ss — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-image-develop.yml (hunk 1 above)
Comment (supertoto29) [reply .github/workflows/docker-image-develop.yml:124]: was kept if @0x72D0 needs it

Comment (supertoto29) [reply .github/workflows/docker-image-develop.yml:124]: Will not work if its uncommented

File: .github/workflows/docker-image-feature.yml (hunk 2 above)
Comment (supertoto29) [reply .github/workflows/docker-image-feature.yml:124]: was kept if @0x72D0 needs it

File: .github/workflows/docker-image-fix.yml (hunk 3 above)
Comment (supertoto29) [reply .github/workflows/docker-image-fix.yml:124]: was kept if @0x72D0 needs it

File: .github/workflows/docker-image-master.yml (hunk 4 above)
Comment (supertoto29) [reply .github/workflows/docker-image-master.yml:133]: was kept if @0x72D0 needs it

**Review by Kam0ss — APPROVED** [review]
Good, but we must not forget the L4T part.

**Pull request merged**